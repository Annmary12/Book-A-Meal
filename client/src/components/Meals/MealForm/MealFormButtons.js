import React from 'react';
import PropTypes from 'prop-types';
import styles from './meal-form.scss';

const ModalButtons = ({ isSubmitting, handleSubmit, handleClose }) => (
  <div className={styles.btns}>
    <button
      type="button"
      disabled={isSubmitting ? 'disabled' : ''}
      className={styles.successBtn}
      onClick={handleSubmit}
    >
      {isSubmitting ? 'Please wait..' : 'Save'}
    </button>
    <button
      type="button"
      className={styles.closeModalBtn}
      onClick={handleClose}
    >
      Cancel
    </button>

  </div>
);

ModalButtons.propTypes = {
  isSubmitting: PropTypes.bool.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  handleClose: PropTypes.func.isRequired
};

export default ModalButtons;
