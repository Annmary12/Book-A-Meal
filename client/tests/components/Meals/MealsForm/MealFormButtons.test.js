import React from 'react';
import MealFormButtons from '../../../../src/components/Meals/MealForm/MealFormButtons';

describe('MealFormButtons component', () => {
  it('should match screenshot if isSubmitting is true', () => {
    const wrapper = shallow(
      <MealFormButtons
        isSubmitting={true}
        handleClose={() => {}}
        handleDelete={() => {}}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
  it('should match screenshot if isSubmitting is false', () => {
    const wrapper = shallow(
      <MealFormButtons
        isSubmitting={false}
        handleClose={() => {}}
        handleDelete={() => {}}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});