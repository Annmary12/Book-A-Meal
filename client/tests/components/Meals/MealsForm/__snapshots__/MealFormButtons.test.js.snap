// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MealFormButtons component should match screenshot if isSubmitting is false 1`] = `
<div
  className="btns"
>
  <button
    className="successBtn"
    disabled=""
    type="button"
  >
    Save
  </button>
  <button
    className="closeModalBtn"
    onClick={[Function]}
    type="button"
  >
    Cancel
  </button>
</div>
`;

exports[`MealFormButtons component should match screenshot if isSubmitting is true 1`] = `
<div
  className="btns"
>
  <button
    className="successBtn"
    disabled="disabled"
    type="button"
  >
    Please wait..
  </button>
  <button
    className="closeModalBtn"
    onClick={[Function]}
    type="button"
  >
    Cancel
  </button>
</div>
`;
